### ISSUE TITLE
kubebuilder init fails when using release candidates of golang

### ISSUE THREAD

**Issue by roberthbailey** [kubernetes-sigs/kubebuilder#439]
I'm not sure why kubebuilder cares what version of go I'm using, but it doesn't like the fact that I'm on a release candidate:

```
$ go version
go version go1.11rc2 linux/amd64
$ kubebuilder version
Version: version.Version{KubeBuilderVersion:"1.0.4", KubernetesVendor:"1.11", GitCommit:"f954dc7cff7e6c94d54139f936bb9d42314ad63e", BuildDate:"2018-09-11T21:08:31Z", GoOs:"unknown", GoArch:"unknown"}
$ kubebuilder init --domain sigs.k8s.io --license apache2 --owner "The Kubernetes Authors" --dep=false
2018/10/12 11:09:34 Invalid go version "1.11rc2": Invalid Semantic Version
```

Since this is how golang releases are officially labeled (see https://groups.google.com/d/msg/golang-announce/zb6p0aFMBJM/Ol5Qj2YKDgAJ) the parser should be more lenient instead of crashing.

**Comment by droot** [comment]
This is fixed with 1.0.5 release

**Comment by roberthbailey** [comment]
Thanks!
